openstax-poland/adaptarr-front
# Full process creation validation

TODO:
- [x] Check if this is actually displaying all of those errors.

## Testing

List of things which are checked for when saving process.
- Verify that name for process exists.
- Verify there are no empty or duplicate names for slots.
- Verify that starting step exists and have links.
- Verify that slots and steps exists
- Verify there are no empty or duplicate names for steps.
- Slots in steps have to be filled.
- Verify that each propose-changes step is followed by an accept-changes step.
- Verify that each accept-changes step is preceded only by propose-changes steps.
- Verify if there is finish step.
- Verify there are no empty or duplicate names for links in steps.
- Verify all steps are reachable from the initial step.
- Verify there's a path from every step to a final step.

## How to check:
ex. for `Verify that name for process exists.` - create new process but don't specify it's name - after clicking `Create process` error should appear.
